contrib/devtools/update-rust-hashes.py

```
@@ -0,0 +1,130 @@
+def download_and_hash(url: str, dest: Path) -> str:
+    hasher = hashlib.sha256()
+    if dest.exists():
+        print(f"  Using existing {dest.name}")
+        with open(dest, "rb") as f:
+            while chunk := f.read(8192):
+                hasher.update(chunk)
+        return hasher.hexdigest()
+
+    print(f"  Downloading {dest.name}")
+    dest.parent.mkdir(parents=True, exist_ok=True)
+    with urllib.request.urlopen(url) as response:
```

Ruff (0.14.14)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)